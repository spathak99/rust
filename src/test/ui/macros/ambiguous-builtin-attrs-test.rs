// compile-flags:--test

#![feature(decl_macro, test)]

extern crate test;

macro test() {}

#[test] //~ ERROR `test` is ambiguous
fn test() {}

macro bench() {}

#[bench] //~ ERROR `bench` is ambiguous
fn bench(b: &mut test::Bencher) {}

fn main() {}
