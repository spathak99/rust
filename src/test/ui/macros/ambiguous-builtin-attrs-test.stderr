error[E0659]: `test` is ambiguous
  --> $DIR/ambiguous-builtin-attrs-test.rs:9:3
   |
LL | #[test] //~ ERROR `test` is ambiguous
   |   ^^^^
   |
note: `test` could refer to the name defined here
  --> $DIR/ambiguous-builtin-attrs-test.rs:7:1
   |
LL | macro test() {}
   | ^^^^^^^^^^^^^^^
note: `test` could also refer to the name defined here
  --> $DIR/ambiguous-builtin-attrs-test.rs:9:3
   |
LL | #[test] //~ ERROR `test` is ambiguous
   |   ^^^^

error[E0659]: `bench` is ambiguous
  --> $DIR/ambiguous-builtin-attrs-test.rs:14:3
   |
LL | #[bench] //~ ERROR `bench` is ambiguous
   |   ^^^^^
   |
note: `bench` could refer to the name defined here
  --> $DIR/ambiguous-builtin-attrs-test.rs:12:1
   |
LL | macro bench() {}
   | ^^^^^^^^^^^^^^^^
note: `bench` could also refer to the name defined here
  --> $DIR/ambiguous-builtin-attrs-test.rs:14:3
   |
LL | #[bench] //~ ERROR `bench` is ambiguous
   |   ^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0659`.
